### ISSUE TITLE
[1.10 pre-release] Perms crash

### ISSUE THREAD

**Issue by perry-mitchell** [buttercup/buttercup-mobile#184]
The following occurs when running the app for the first time:

```
09-12 19:08:59.664 12361 12361 I System.out: Checking perms
09-12 19:08:59.671 12361 12361 I System.out: Can't draw overlays
09-12 19:08:59.672  1597  2644 I ActivityManager: START u0 {act=android.settings.action.MANAGE_OVERLAY_PERMISSION dat=package:com.buttercup cmp=com.android.settings/.Settings$AppDrawOverlaySettingsActivity} from uid 10085
09-12 19:08:59.676  1597  2644 V WindowManager: setAppStartingWindow: token=Token{bff3506 ActivityRecord{32a13e1 u0 com.android.settings/.Settings$AppDrawOverlaySettingsActivity t23}} pkg=com.android.settings transferFrom=Token{e16cf39 ActivityRecord{9e1bb00 u0 com.buttercup/.MainActivity t23}} newTask=false taskSwitch=false processRunning=false allowTaskSnapshot=true
09-12 19:08:59.676  1597  2644 V WindowManager: Checking theme of starting window: 0x7f1301c4
09-12 19:08:59.676  1597  2644 V WindowManager: Translucent=false Floating=false ShowWallpaper=false
09-12 19:08:59.676  1597  2644 V WindowManager: Moving existing starting Window{1f18fc4 u0 Splash Screen com.buttercup} from AppWindowToken{1efc97e token=Token{e16cf39 ActivityRecord{9e1bb00 u0 com.buttercup/.MainActivity t23}}} to AppWindowToken{67597c7 token=Token{bff3506 ActivityRecord{32a13e1 u0 com.android.settings/.Settings$AppDrawOverlaySettingsActivity t23}}}
09-12 19:08:59.676  1597  2644 V WindowManager: Removing starting Window{1f18fc4 u0 Splash Screen com.buttercup} from AppWindowToken{1efc97e token=Token{e16cf39 ActivityRecord{9e1bb00 u0 com.buttercup/.MainActivity t23}}}
09-12 19:08:59.676  1597  2644 V WindowManager: Nulling last startingData
09-12 19:08:59.687  1597  1875 I ActivityManager: START u0 {act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.android.packageinstaller cmp=com.android.packageinstaller/.permission.ui.GrantPermissionsActivity (has extras)} from uid 10085
09-12 19:08:59.689  1597  1875 V WindowManager: setAppStartingWindow: token=Token{1530d1d ActivityRecord{f3a71f4 u0 com.android.packageinstaller/.permission.ui.GrantPermissionsActivity t23}} pkg=com.android.packageinstaller transferFrom=Token{bff3506 ActivityRecord{32a13e1 u0 com.android.settings/.Settings$AppDrawOverlaySettingsActivity t23}} newTask=false taskSwitch=false processRunning=true allowTaskSnapshot=true
09-12 19:08:59.689  1597  1875 V WindowManager: Checking theme of starting window: 0x7f1000a9
09-12 19:08:59.689  1597  1875 V WindowManager: Translucent=false Floating=true ShowWallpaper=false
09-12 19:08:59.700 12361 12388 W unknown:InspectorPackagerConnection: Couldn't connect to packager, will silently retry
09-12 19:08:59.728  1597  1615 I ActivityManager: Start proc 12391:com.android.settings/1000 for activity com.android.settings/.Settings$AppDrawOverlaySettingsActivity
09-12 19:08:59.745 12391 12391 W zygote  : Unexpected CPU variant for X86 using defaults: x86
09-12 19:08:59.793  1333  1358 E SurfaceFlinger: ro.sf.lcd_density must be defined as a build property
09-12 19:08:59.793 12361 12405 D OpenGLRenderer: HWUI GL Pipeline
09-12 19:08:59.809  1597  2445 W WindowManager: Failed looking up window callers=com.android.server.wm.WindowManagerService.addWindow:1166 com.android.server.wm.Session.addToDisplay:190 android.view.IWindowSession$Stub.onTransact:124
09-12 19:08:59.809  1597  2445 W WindowManager: Attempted to add window with token that is not a window: null.  Aborting.
09-12 19:08:59.813 12361 12361 D AndroidRuntime: Shutting down VM
09-12 19:08:59.814 12361 12361 E AndroidRuntime: FATAL EXCEPTION: main
09-12 19:08:59.814 12361 12361 E AndroidRuntime: Process: com.buttercup, PID: 12361
09-12 19:08:59.814 12361 12361 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window -- token null is not valid; is your activity running?
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:764)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:356)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:92)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.widget.PopupWindow.invokePopup(PopupWindow.java:1429)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.widget.PopupWindow.showAtLocation(PopupWindow.java:1203)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.widget.PopupWindow.showAtLocation(PopupWindow.java:1170)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at com.facebook.react.devsupport.DevLoadingViewController.showInternal(DevLoadingViewController.java:150)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at com.facebook.react.devsupport.DevLoadingViewController.access$000(DevLoadingViewController.java:32)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at com.facebook.react.devsupport.DevLoadingViewController$1.run(DevLoadingViewController.java:54)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:769)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6540)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
09-12 19:08:59.814 12361 12361 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
09-12 19:08:59.826  1597  1620 I ActivityManager: Showing crash dialog for package com.buttercup u0
```

Android 8 w/ Google Play services, fresh install, emulator.

**Comment by se1exin** [comment]
Not sure if this question has been asked, but is the `SYSTEM_ALERT_WINDOW` (display over other apps) permission even required? I know that it is required in debug for react native debugging (error boxes etc), but if it isn't needed for the production build can we just remove the permission altogether?

A few interesting links:
- Using `tools:node="remove"` to strip unneeded RN permissions: https://facebook.github.io/react-native/docs/removing-default-permissions
- Discussion with workarounds to unneeded/unwanted permissions get merged back in during build, and only removing during prod build (vs debug): https://github.com/facebook/react-native/issues/5886

**Comment by se1exin** [comment]
I've just tested out running the app without the permission, and everything seems OK. I've created PR #185 if you want to go ahead with removing the permission.

**Comment by perry-mitchell** [comment]
Seems to have been fixed! Thanks @se1exin!

**Comment by perry-mitchell** [comment]
Should be available in 1.10.0
